src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java

```diff
@@ -76,14 +74,22 @@ public ReadsSparkSource(final JavaSparkContext ctx, final ValidationStringency v
 
 
     public static class SplitSortingSamInputFormat extends AnySAMInputFormat{
-
+        @SuppressWarnings("unchecked")
         @Override
         public List<InputSplit> getSplits(JobContext job) throws IOException {
             final List<InputSplit> splits = super.getSplits(job);
-            splits.sort((a, b) -> {
-                FileSplit fa = (FileSplit) a, fb = (FileSplit) b;
-                return fa.getPath().compareTo(fb.getPath());
-            });
+
+
+            if( splits.stream().allMatch(split -> split instanceof FileVirtualSplit || split instanceof FileSplit)) {
+                splits.sort(Comparator.comparing(split -> {
+                    if (split instanceof FileVirtualSplit) {
```

Comment:
You mix the virtual and non-virtual splits here. This makes me uncomfortable as I don't know that its guaranteed either is going to return the same thing for getPath(). Could you use the filename instead? As that seems to be what is being sorted here anyway.